### PR TITLE
bug63394 output 1 when the file doesn't exist or can't be opened

### PR DESCRIPTION
**Pull request by shaqianqian**

### CODE REVIEW

File: src/core/org/apache/jmeter/JMeter.java [hunk 1]
```diff
@@ -997,6 +997,7 @@ private void runNonGui(String testFile, String logFile, boolean remoteStart, Str
             File f = new File(testFile);
             if (!f.exists() || !f.isFile()) {
                 println("Could not open " + testFile);
+                System.exit(1);
```
Comment (pmouawad) [inline src/core/org/apache/jmeter/JMeter.java:1000]: Hello @shaqianqian , this condition is good but not enough. A test plan opening can fail if file exists but it relies on 3rd party plugin that is not installed.

Comment (shaqianqian) [reply src/core/org/apache/jmeter/JMeter.java:1000]: Hello,
About the third party plugin, it's something like this? [a example](https://jmeter-plugins.org/wiki/TestPlanCheckTool/)
Thanks

Comment (shaqianqian) [reply src/core/org/apache/jmeter/JMeter.java:1000]: Hello, 
The new commit is correct? I have tested it by the "DummySampler". 
____________________________________________________________________________

>  Detail:com.thoughtworks.xstream.converters.ConversionException:
> ---- Debugging information ----
> cause-exception     : com.thoughtworks.xstream.converters.ConversionException
> cause-message       :
> first-jmeter-class  : org.apache.jmeter.save.converters.HashTreeConverter.unmarshal(HashTreeConverter.java:67)
> class               : org.apache.jmeter.save.ScriptWrapper
> required-type       : org.apache.jmeter.save.ScriptWrapper
> converter-type      : org.apache.jmeter.save.ScriptWrapperConverter
> path                : /jmeterTestPlan/hashTree/hashTree/hashTree/kg.apc.jmeter.samplers.DummySampler
> line number         : 28
> version             : 5.2-SNAPSHOT.20190506
> ……………………
> 1

**Comment by shaqianqian** [comment]
I have tested jmeter --server/-s, the result is like this. When there is un exception, the result is the first one , or not is the second one.

> $ bin/jmeter -s -Djava.rmi.server.hostname=10.0.1.111
> Server failed to start: java.rmi.server.ExportException: Listen failed on port: 0; nested exception is:
>         java.io.FileNotFoundException: rmi_keystore.jks (Le fichier sp▒cifi▒ est introuvable)
> An error occurred: Listen failed on port: 0; nested exception is:
>         java.io.FileNotFoundException: rmi_keystore.jks (Le fichier sp▒cifi▒ est introuvable)
----------------------------------------------------------------------------------------------------------
> $ bin/jmeter -s -Djava.rmi.server.hostname=10.0.1.111
> Created remote object: UnicastServerRef2 [liveRef: [endpoint:[10.0.1.111:55914](local),objID:[-586845f4:16a8caa5ce8:-7fff, 7745514337969577424]]]
.....

File: src/core/org/apache/jmeter/JMeter.java (hunk 1 above)
Comment (shaqianqian) [reply src/core/org/apache/jmeter/JMeter.java:1000]: I have tested jmeter --server/-s, the result is like this. When there is un exception, the result is the first one , or not is the second one.

> $  bin/jmeter  -s -Djava.rmi.server.hostname=10.0.1.ééé -n -t test.jmx -l log; echo $?
> Created remote object: UnicastServerRef2 [liveRef: [endpoint:[10.0.1.▒▒▒:59805](local),objID:[27993ad6:16a8d5d35c2:-7fff, 3041592823253102291]]]
> Server failed to start: java.rmi.RemoteException: Cannot start. Unable to get local host IP address.; nested exception is:
>         java.net.UnknownHostException: 10.0.1.▒▒▒
> An error occurred: Cannot start. Unable to get local host IP address.; nested exception is:
>         java.net.UnknownHostException: 10.0.1.▒▒▒
> 1

----------------------------------------------------------------------------------------------------------
> $ bin/jmeter -s -Djava.rmi.server.hostname=10.0.1.111
> Created remote object: UnicastServerRef2 [liveRef: [endpoint:[10.0.1.111:55914](local),objID:[-586845f4:16a8caa5ce8:-7fff, 7745514337969577424]]]
.....

Comment (shaqianqian) [reply src/core/org/apache/jmeter/JMeter.java:1000]: For jmeter --runremote/-r
__________________________________________

> $  bin/jmeter -r -n -t test.jmx -l log; echo $?
> Error in NonGUIDriver java.lang.IllegalArgumentException: Problem loading XML from:'C:\bug63394\jmeter\test.jmx'.
> Cause:
> CannotResolveClassException: kg.apc.jmeter.samplers.DummySampler
> 
>  Detail:com.thoughtworks.xstream.converters.ConversionException:
> ---- Debugging information ----
> cause-exception     : com.thoughtworks.xstream.converters.ConversionException
> cause-message       :
> first-jmeter-class  : org.apache.jmeter.save.converters.HashTreeConverter.unmarshal(HashTreeConverter.java:67)
> class               : org.apache.jmeter.save.ScriptWrapper
> required-type       : org.apache.jmeter.save.ScriptWrapper
> converter-type      : org.apache.jmeter.save.ScriptWrapperConverter
> path                : /jmeterTestPlan/hashTree/hashTree/hashTree/kg.apc.jmeter.samplers.DummySampler
> line number         : 28
> version             : 5.2-SNAPSHOT.20190506
> ..........................................................................
> 1

**Pull request merged**